src/julia.h

```diff
@@ -65,6 +65,23 @@
 #  define JL_THREAD_LOCAL
 #endif
 
+// Duplicated from options.h
```

Comment:
This seems oddly terrible. Can we delete it from there?

Reply:
Maybe put it in dtypes.h? We've seem to have accumulated a bunch of these kinds of things there.

Reply:
Actually, for the moment, there's probably no reason not to just put this here. Does raise the question of why jl_task_t is defined in the public header in the first place. We're never passing it by value, so it should probably move to the internals header, but we can do that separately.

Reply:
Actually, let me just do that first.

Reply:
Nevermind, we need the layout information in the public header because locks depend on it for ABI reasons. We should disentangle all of that at some point, but for now, I'll just leave this here and delete it elsewhere.